### ISSUE TITLE
[Question]: How to gracefully disconnect from a voice channel

### ISSUE THREAD

**Issue by nullpat** [PerformanC/FastLink#13]
### FastLink version

2.4.1-rc.1

### Description

```javascript
player.destroy();
```
this leaves the bot connected to a voice channel
```javascript
  const payload = {
    op: 4,
    d: {
      guild_id: guildId,
      channel_id: null,
      self_mute: false,
      self_deaf: false,
    },
  };
  client.guilds.cache.get(guildId).shard.send(payload);
```

setting channel_id here to null successfully disconnects the bot from the voice channel, however the bot has a 50% chance of blowing up with `Uncaught TypeError TypeError: Cannot read properties of undefined (reading 'queue')
    at update (\node_modules\@performanc\fastlink\index.js:248:29)` when trying to add a song to the queue after creating and connecting the player again. Feels like there is a race condition somewhere

1. player.destroy();
2. send disconnectPayload (channel_id set to null)
3. player.createPlayer();
4. player.connect
5. add something to queue (sometimes blows up here on first pass, sometimes i repeat 1-5 again then it blows up)

How would you recommend disconnecting the bot from a voice channel? I'm not sure if im missing something obvious

Thank you for your time.

**Comment by nullpat** [comment]
additionally `FastLink.player.getPlayers` does not work with lavalink
````
{timestamp: 1731625192214, status: 404, error: 'Not Found', message: 'No endpoint GET /v4/sessions.', path: '/v4/sessions'}
````

updating the getPlayers endpoint to `/v4/sessions/${Nodes[node].sessionId}/players` works

**Comment by ThePedroo** [comment]
To disconnect you can call .connect(null). About the second issue, I'll take a look further.

**Comment by ThePedroo** [comment]
Are you sure you are calling `.createPlayer()` after `.destroy()` and `.update()` after `.createPlayer()`?

**Comment by nullpat** [comment]
Yes, the easiest way for you to reproduce will be to add the following command to your example bot script

```javascript
  if (commandName === "disconnect") {
    const payload = {
      op: 4,
      d: {
        guild_id: message.guild.id,
        channel_id: null,
        self_mute: false,
        self_deaf: false,
      },
    };
    client.guilds.cache.get(message.guild.id).shard.send(payload);
    message.channel.send("Disconnected.");
    return;
  }
```

1. !play 123
2. !stop
3. !disconnect
4. !play 123

Thanks again

**Comment by ThePedroo** [comment]
You should se .connect(null, ...) instead also.

**Comment by nullpat** [comment]
At present it doesn't allow null

```javascript
  connect(voiceId, options, sendPayload) {  
    if (!voiceId) throw new Error('No voiceId provided.')
```

**Comment by ThePedroo** [comment]
https://github.com/PerformanC/FastLink/blob/main/index.js#L195 ?

**Comment by nullpat** [comment]
looks like npm is serving an old version
`npm i @performanc/fastlink`
```
    "dependencies": {
        "@performanc/fastlink": "^2.4.1-rc.1",```

**Comment by ThePedroo** [comment]
I'll update it soon. I recommend to use GH version for now.
